lnd_test.go

```diff
@@ -16,109 +13,181 @@ import (
 	"github.com/roasbeef/btcd/wire"
 	"github.com/roasbeef/btcrpcclient"
 	"github.com/roasbeef/btcutil"
+	"google.golang.org/grpc"
+	"testing"
+	"github.com/go-errors/errors"
 )
 
-type lndTestCase func(net *networkHarness, t *testing.T)
+// CT is needed for:
+// - have uniform way of handling panic and fatal error from test cases.
+// - have ability to properly wrap errors in order to see stack trace.
+// - have nice and elegant way to handle lnd process errors in one
+// select structure with test cases.
+type CT struct {
+	*testing.T
+
+	// Channel for sending retransmitted panic errors and fatal error which
+	// happens in test case.
+	errChan chan interface{}
+}
+
+func NewCT(t *testing.T) *CT {
+	return &CT{t, nil}
+}
+
+func (ct *CT) Error(err error) {
+	if ct.errChan != nil {
+		ct.errChan <- errors.Wrap(err, 1).ErrorStack()
+		ct.FailNow()
+	} else {
+		ct.Fatal("can't sen error when test isn't running")
```

Comment:
I added this check in order to exclude the cases when developer might send error in `nil` channel (channel becomes not nil only during test case execution).